### PR TITLE
[android] update base64 decoding options

### PR DESCRIPTION
**Pull request by jarvisluong**
# Why

This PR is for aligning the base64 decoding output format for Android, iOS and Web in certain modules: ImagePicker, Camera, ImageManipulator, Print and SecureStore

Linked issue: https://github.com/expo/expo/issues/7837

# How

I was using ImageManipulator and noticed that the output of base64 result was not similar from Android comparing iOS and web, and learned that the default decoding options for Android was the cause.

# Test Plan

I updated the e2e test to check for the appearance of newline and \r character

### CODE REVIEW

File: apps/test-suite/tests/Camera.js [hunk 1]
```diff
@@ -94,7 +94,9 @@ export async function test(t, { setPortalChild, cleanupPortal }) {
         t.expect(picture.exif).toBeDefined();
       });
 
-      t.it('returns Base64 only if requested', async () => {
+      t.it(
+        'returns Base64 only if requested, and base64 result does not contain newline character',
+        async () => {
```
Comment (bbarthec) [inline apps/test-suite/tests/Camera.js:99]:
```suggestion
      t.it(
        'this long description is forcing us to move testing method signature to the next line, but let's keep method body distinguishable by keeping indentation anyway',
        async () => {
          await mountAndWaitFor(<Camera ref={refSetter} style={style} />);
          let picture = await instance.takePictureAsync({ base64: false });	
          t.expect(picture).toBeDefined();
          ...
        }
      )
```

File: packages/expo-camera/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 ### 🛠 Breaking changes
 
+* Android: the base64 output will not contain newline and special character (\n, \r)
```
Comment (bbarthec) [inline packages/expo-camera/CHANGELOG.md:7]:
```suggestion
- The base64 output will no longer contain newline and special character (`\n`, `\r`) on Android. ([#7841](https://github.com/expo/expo/pull/7841) by [@jarvisluong](https://github.com/jarvisluong))
```

File: packages/expo-image-manipulator/CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,8 @@
 
 ### 🛠 Breaking changes
 
+* Android: the base64 output will not contain newline and special character (\n, \r)
```
Comment (bbarthec) [inline packages/expo-image-manipulator/CHANGELOG.md:7]:
```suggestion
- The base64 output will no longer contain newline and special character (`\n`, `\r`) on Android. ([#7841](https://github.com/expo/expo/pull/7841) by [@jarvisluong](https://github.com/jarvisluong))
```

File: packages/expo-image-picker/CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,8 @@
 
 ### 🛠 Breaking changes
 
+* Android: the base64 output will not contain newline and special character (\n, \r)
```
Comment (bbarthec) [inline packages/expo-image-picker/CHANGELOG.md:7]:
```suggestion
- The base64 output will no longer contain newline and special character (`\n`, `\r`) on Android. ([#7841](https://github.com/expo/expo/pull/7841) by [@jarvisluong](https://github.com/jarvisluong))
```

File: packages/expo-secure-store/CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,8 @@
 
 ### 🛠 Breaking changes
 
+* Android: the base64 output will not contain newline and special character (\n, \r)
```
Comment (bbarthec) [inline packages/expo-secure-store/CHANGELOG.md:7]:
```suggestion
- The base64 output will no longer contain newline and special character (`\n`, `\r`) on Android. ([#7841](https://github.com/expo/expo/pull/7841) by [@jarvisluong](https://github.com/jarvisluong))
```

File: apps/test-suite/tests/Camera.js [hunk 6]
```diff
@@ -95,8 +95,14 @@ export async function test(t, { setPortalChild, cleanupPortal }) {
...
       t.it(
-        'returns Base64 only if requested, and base64 result does not contain newline character',
+        'this long description is forcing us to move testing method signature to the next line, but let's keep method body distinguishable by keeping indentation anyway',
         async () => {
+          await mountAndWaitFor(<Camera ref={refSetter} style={style} />);
+          let picture = await instance.takePictureAsync({ base64: false });	
+          t.expect(picture).toBeDefined();
+          ...
+        }
+      )
```
Comment (bbarthec) [inline apps/test-suite/tests/Camera.js]: Well, I'm afraid that is not what I was suggesting 😅
Sorry for using `suggestion` block for this one ... 😞
I just wanted to present my idea 😉
Can you please fix this one?

Comment (jarvisluong) [reply apps/test-suite/tests/Camera.js]: ah 😅 got it

**Review by bbarthec — APPROVED** [review]

File: packages/expo-print/CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,8 @@
 
 ### 🛠 Breaking changes
 
+* Android: the base64 output will not contain newline and special character (\n, \r)
```
Comment (lukmccall) [inline packages/expo-print/CHANGELOG.md:7]:
```suggestion
- The base64 output will no longer contain newline and special character (`\n`, `\r`) on Android. ([#7841](https://github.com/expo/expo/pull/7841) by [@jarvisluong](https://github.com/jarvisluong))
```
Please follow the changelog convection.

**Review by lukmccall — APPROVED** [review]
LGTM 🎉
Just one comment ;)

File: packages/expo-print/CHANGELOG.md (hunk 7 above)
Comment (jarvisluong) [reply packages/expo-print/CHANGELOG.md:7]: 😄

**Comment by sofianear** [comment]
Hello, 
I have a question about `expo-secure-store` in __sdk 38__ and I hope it's the right place to ask it.
I am currently using `expo-secure-store` under expo __sdk 37__ and I'd like to upgrade to __sdk 38__.
It is mentioned in the CHANGELOG of the __sdk 38__ that this pull request has introduced breaking changes: 
>The base64 output will no longer contain newline and special character

So my question is: how will these changes affect my usage of `expo-secure-store` module and what should I do to fix the issues if any.
I have actually went through the API reference of `expo-secure-store`  (for both __sdk 37__ and __sdk 38__) and I haven't seen anything about base64 encoding/decoding there. So I really don't know what to do, and I don't want to upgrade to __sdk 38__ only to find issues that I don't know how to fix.

**Comment by jarvisluong** [comment]
@sofianear As far as I understand, if you don't care about newline characters and /r character you are good to go. Do you send the base64 output to the server? I believe most base64 decoder now handle just fine a standard base64 output without special characters in it.

**Comment by sofianear** [comment]
@sofianear 
I am storing a string in the local storage using `expo-secure-store`. I also extract it and send it to a remote server. So I suppose that when I extract the string from local storage, it is decoded and therefore not in Base64. Is that right ?

**Comment by jarvisluong** [comment]
@sofianear That should be right 🙂

**Comment by sofianear** [comment]
@jarvisluong  Thank you.

**Pull request merged**